packages/public/inspection-capture-web/src/hooks/useCameraConfig.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+import { CameraFacingMode, CameraResolution, CompressionFormat } from '@monkvision/camera-web';
+
+export function useCameraConfig() {
```

Comment:
Okay so I think you created this hook to mimic how I implemented it in the TestView component in the test app. However I did this just to test the component. In reality we don't need all of this.

I think you can remove this hook and have all the camera config be passed as props to the PhotoCapture component (for now, maybe we will change this later)

Reply:
I created this hook to also get the CameraDimensions which is used in the hook SightPreviewStyle. 
The goal is to assign the right aspectRatio for the SightOverlay. (Instead of '16/9' that I have implemented previously)

Should we keep this useCameraConfig for that?

Reply:
Oooooh okay I misread the hook, I think your approach is wrong, I'll explain it to you by voice later it will be easier

Reply:
Ooooh okay so I misread the hook my bad, I think your approach is wrong. Let me explain to you how the Camera works internally so that you'll get a better grasp at how things are working :

- We pass some `CameraConfig` properties as props to the `Camera` component. These properties allow us to configure some properties of the camera such as the resolution, the orientation (front / rear camera) etc.
- When the `Camera` component receives these `CameraConfig` values, it converts them into `MediaConstraints` using the `getMediaConstraints` function (I will explain to you what are `MediaConstraints` later).
- Then, the `Camera` component will take these `MediaConstraints` and pass them to the `useUserMedia` hook. This hook is simply a wrapper hook, that wraps the `getUserMedia` function and adds React logic like state management etc to it.
- The `getUserMedia` function is a Vanilla JavaScript function available on all browsers ([here is the official doc](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia)). This function does the following : 
  - It accepts `MediaConstraints` as parameters. These constraints are simply Media requirements that we ask to the browser (for instance, we ask the browser "_hey, can you give me access to the front camera, with a 1080p resolution please?_")
  - It then asks the browser to give us the `MediaStream` object (a media stream for the device, like a camera stream or audio stream) and it returns the `MediaStream`
- The `Camera` component then binds the resulting `MediaStream` (which corresponds to the device camera stream) to an HTML `video` element to display the camera stream onto the screen.

Now that you understand how the `Camera` component works, note one things : the `MediaConstraints` are only "requirements" that we ask to the browser, but they are not necessarily the EXACT properties of the resulting stream that we will get. For instance, what happens if we ask the browser for a 4K UHD camera stream when the device only has a 1080p camera? Or when we ask for a rear camera when the device is a laptop and it only has a front facing camera? Well, the answer is : it depends on the browser 😂 Sometimes, the browser will crop and scale the original camera feed to make it match the requirements, sometimes it will completely ignore our requirements and directly give us the original camera feed. Basically the take away from all of this is : the specs we pass to the `Camera` component are only "requirements", but not necessarily the resulting properties.

Reply:
The `Camera` component should pass the stream dimensions in the camera handle to the HUD :

- Dans `CameraHUD.types.ts` : rajouter un champ `streamDimensions` à l'interface `CameraHandle`
- Les dimensions sont déjà retournées par le hook `useCameraPreview` (dans le champ `dimensions`), il suffit à la Camera de passer ces dimensions à la handle

Reply:
I kept `dimensions` instead of `streamDimensions` for syntax consistency:

`<HUDComponent
      handle={{ takePicture, error, retry, isLoading, dimensions }}
      cameraPreview={cameraPreview}
    />`
    
  if streamDimensions is more explicit, let me know, I will modify it.